contents/Documentation/Optics.playground/Pages/Composition.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,105 @@
+/*
+ layout: docs
+ title: Composition
+ */
+// nef:end
+// nef:begin:hidden
+import Bow
+import BowOptics
+// nef:end
+/*:
+ # Composition
+ 
+ Optics can compose to create more powerful optics and access deeply nested data structures in an seamless manner. The result of the composition of the 8 pairs of optics that are included in Bow is shown in the following table. Notice that not all combinations are possible for composition, and that some times the resulting optic is not the same as the optics that were composed.
+ 
+ |               | **Iso**   | **Lens**  | **Prism** | **Optional** | **Getter** | **Setter** | **Fold** | **Traversal** |
+ | ------------- | --------- | --------- | --------- | ------------ | ---------- | ---------- | -------- | ------------- |
+ | **Iso**       | Iso       | Lens      | Prism     | Optional     | Getter     | Setter     | Fold     | Traversal     |
+ | **Lens**      | Lens      | Lens      | Optional  | Optional     | Getter     | Setter     | Fold     | Traversal     |
+ | **Prism**     | Prism     | Optional  | Prism     | Optional     | 🚫         | Setter     | Fold     | Traversal     |
+ | **Optional**  | Optional  | Optional  | Optional  | Optional     | Fold       | Setter     | Fold     | Traversal     |
+ | **Getter**    | Getter    | Getter    | 🚫        | 🚫           | Getter     | 🚫         | Fold     | 🚫            |
+ | **Setter**    | Setter    | Setter    | Setter    | Setter       | 🚫         | Setter     | 🚫       | Setter        |
+ | **Fold**      | Fold      | Fold      | Fold      | Fold         | Fold       | 🚫         | Fold     | Fold          |
+ | **Traversal** | Traversal | Traversal | Traversal | Traversal    | Fold       | Setter     | Fold     | Traversal     |
+ 
+ Each optic can be composed using its instance method `compose`, which is overloaded to accept all possible combinations with other optics. Besides, for the sake of simplicity, operator `+` can also be used to compose any two optics.
+ 
+ ## Example of composition: N-ary Tree
+ 
+ An N-ary tree is a tree where each node can have any number of branches. We can model it like:
+ */
+enum NTree<A> {
+    case leaf(A)
+    indirect case node(A, branches: NEA<NTree<A>>)
+}
+/*:
+ That is, we can have a leaf with an associated value, or a node with an associated value and a `NonEmptyArray` of branches (it must have at least one, otherwise it would be a leaf).
+ 
+ Let's imagine that we would like to combine all values of the nodes at level `m`. Stop for a moment and think how you would do it without optics. It does not have a trivial solution, right? Let's see if we can leverage the power of optics to do this.
+ 
+ We can start by trying to access the branches of a node. Since `NTree` is a sum type, the optic that we need to use is a Prism. We can use `AutoPrism` to get the `Prism` for the node side of the `NTree`:
+ */
+extension NTree: AutoPrism {}
+
+func nodePrism<A>() -> Prism<NTree<A>, (A, NEA<NTree<A>>)> {
```

Comment:
maybe it could be nice to make emphasis on why do we need to apply pattern matching (this is explained in another page and reader could start by this one)

```> Note:  we need to provide a bit of help by providing a pattern matching function that extracts the associated values out of the case we are focusing on
```